namespaces/mson-namespace.md

```diff
@@ -0,0 +1,497 @@
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "member",
+            "content": {
+                "key": {
+                    "element": "string",
+                    "content": "id"
+                },
+                "value": {
+                    "element": "number",
+                    "attributes": {
+                        "default": 0
+                    },
+                    "content": null
+                }
+            }
+        }
+    ]
+}
+```
+
+### One Of
+
+#### MSON
+
+```
+- city
+- One Of
+    - state
+    - province
+```
+
+#### MSON DOM
+
+TODO:
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "member",
+            "content": {
+                "key": {
+                    "element": "string",
+                    "content": "city"
+                }
+            }
+        },
+        {
+            "element": "select",
+            "content": [
+                {
+                    "element": "option",
+                    "content": [
+                        {
+                            "element": "member",
+                            "content": {
+                                "key": {
+                                    "element": "string",
+                                    "content": "state"
+                                }
+                            }
+                        }
+                    ]
+                },
+                {
+                    "element": "option",
+                    "content": [
+                        {
+                            "element": "member",
+                            "content": {
+                                "key": {
+                                    "element": "string",
+                                    "content": "province"
+                                }
+                            }
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+### Mixin
+
+#### MSON
+
+```
+- id
+- Include User
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "extend",
+    "content": [
+        {
+            "element": "object",
+            "content": [
+                {
+                    "element": "member",
+                    "content": {
+                        "key": {
+                            "element": "string",
+                            "content": "id"
+                        }
+                    }
+                }
+            ]
+        }
+        {
+            "element": "ref",
+            "content": "User"
+        }
+    ]
+}
+```
+
+### Named Type
+
+#### MSON
+
+```
+# Address (object)
+
+Description is here! Properties to follow.
+
+## Properties
+
+- street
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "meta": {
+        "id": "Address",
+        "title": "Address",
+        "description": "Description is here! Properties to follow."
```

Comment:
Descriptions, no matter if they are block or inline, seem to me to be attributes of the given element. Adding them as members only works if you define a special way to process the MSON `object` elements that is completely different than Refract `object` elements. In other words, currently, the base Refract `object` MUST be an object or array of members. In MSON, though, if there is an `object`, these special `content` elements should be ignored when converting to JSON.

This is already a rule with regards to `attributes` of an element, since they drop away when converting to JSON. 

Lastly, this seems to be true of all kinds of elements, not just objects. For instance, `string` elements may have block descriptions, but they have simply a string as their content. I may be wrong there.

Hope that makes sense.